helix-term/src/handlers/completion.rs

```diff
@@ -291,6 +300,213 @@ fn request_completion(
+                        Some(1)
+                    } else if matched_path.starts_with("$HOME") {
+                        Some(5)
+                    } else if matched_path.starts_with("${HOME}") {
+                        Some(7)
+                    } else {
+                        None
+                    };
+                    if let Some(start_char) = first_separator_after_home {
+                        let mut path = home.to_owned();
+                        path.push(&matched_path[start_char..]);
+                        path
+                    } else {
+                        matched_path.into()
+                    }
+                } else {
+                    matched_path.into()
+                };
+                PathBuf::from(home_resolved_path)
+            };
+            #[cfg(not(unix))]
+            let mut path = PathBuf::from(matched_path);
+
+            if path.is_relative() {
+                if let Some(doc_path) = doc.path().and_then(|dp| dp.parent()) {
+                    path = doc_path.join(path);
+                } else if let Ok(work_dir) = std::env::current_dir() {
+                    path = work_dir.join(path);
+                }
+            }
+            let ends_with_slash = match matched_path.chars().last() {
+                Some(std::path::MAIN_SEPARATOR) => true,
+                None => return None,
+                _ => false,
+            };
+            // check if there are chars after the last slash, and if these chars represent a directory
+            match std::fs::metadata(path.clone()).ok() {
+                Some(m) if m.is_dir() && ends_with_slash => {
+                    Some((PathBuf::from(path.as_path()), None))
+                }
+                _ if !ends_with_slash => path.parent().map(|parent_path| {
+                    (
+                        PathBuf::from(parent_path),
+                        path.file_name().and_then(|f| f.to_str().map(String::from)),
+                    )
+                }),
+                _ => None,
+            }
+        })?;
+
+    // The async file accessor functions of tokio were considered, but they were a bit slower
+    // and less ergonomic than just using the std functions in a separate "thread"
+    let future = tokio::task::spawn_blocking(move || {
+        let Some(read_dir) = std::fs::read_dir(&dir_path).ok() else {
+            return Vec::new();
+        };
+
+        read_dir
+            .filter_map(Result::ok)
+            .filter_map(|dir_entry| dir_entry.metadata().ok().map(|md| (dir_entry.path(), md)))
+            .map(|(path, md)| {
+                let file_name = path
+                    .file_name()
+                    .unwrap_or_default()
+                    .to_string_lossy()
+                    .to_string();
+
+                let full_path = path.canonicalize().unwrap_or_default();
+                let full_path_name = full_path.to_string_lossy().into_owned();
```

Comment:
For canonicalization we generally prefer `helix_stdx::path::canonicalize` now (it doesn't expand symlinks like the canonicalize in `std`). For the full path name that we display in the documentation let's pass that canonicalized path through `helix_stdx::path::fold_home_dir` so it displays better

Reply:
Hmm, I wonder though, if a link *should* be shown resolved then or not (because then I think we still need `std::path::canonicalize`), which IMO makes sense but I have no strong opinion here...

Reply:
I think we should try to avoid expanding symlinks in the parent directories as std's canonicalize would do. For an edge case, my `~/src/` directory is a symlink to `/nix/persist/home/michael/src` so canonicalizing a path like `./helix-event` would expand all the way to `/nix/persist/home/michael/src/helix/hx/helix-event`. If the directory entry that we are looking at is a symlink though we should probably follow that link (once?).

Reply:
Yeah it's what I'd expect, but I see your point.

I think following a link once, would result in otherwise inconsistent behavior but I'm not sure. So I'd rather not follow it, and display it as link then, I've pushed a version without following the link for now.